### ISSUE TITLE
Create teacher entity & import initial records

### ISSUE THREAD

**Issue by bnchdrff** [ESDet/esdportal_profile#29]
- [x] @anderoy provide sample data for teacher import
- [x] design basic teacher entity (based on what eric has started in 28-e3entities & specs at https://docs.google.com/document/d/1_tApwQ24lR0xwd0kbgFjqIKfT9eePxWycD2mE27gOnA/edit#heading=h.huh6zdpny8m0)
- [x] write migrate module for teacher data
- [x] merge into master & run live import

**Comment by bnchdrff** [comment]
sample teacher data is in goodrive

**Comment by anderoy** [comment]
@bnchdrff For the purposes of the auto complete function (which will work off of the Title field) we should input the full name into the title field but still add the firstname and lastname fields which will make for better sorting, filtering, and data export. I have resent the csv.

**Comment by bnchdrff** [comment]
* eric's created entity type was `teacher` but it should be `contact` - recreating it from scratch
* just noticed that the bundle wasn't included in my feature -- wat? note to self, `eck_bundle` and `eck_entity_type` are separate feature componenets.
* bad news: after testing with various comibnations of code, i've discovered that these eck entity types & bundles aren't fully exportable as features! this is bad, and sort of a show stopper. we need to either do entity definition in code, or use eck & drop the potential of controlling these new things in code.

**Comment by anderoy** [comment]
What (in layman's terms) is the benefit of having the entity types exportable as features and why is it a show stopper in your mind?

What is lost by not being able to control these things in code?

**Comment by bnchdrff** [comment]
It guarantees that things work if someone messes them up -- there will be custom code relying on these entities acting a certain way, and we need to ensure that they are in a certain state in order for that code to function reliably.

**Comment by anderoy** [comment]
So how are the entities stored, how do they work currently with eck?

**Comment by bnchdrff** [comment]
They're stored in the database according to the entity definition... I'll be working through this more this evening & will figure out a solution!

**Comment by bnchdrff** [comment]
reference material:

* http://lookalive.co.uk/blog/drupal-entity-reference-how-make-custom-entity-reference-selection-handler
* https://www.drupal.org/node/878784

**Comment by bnchdrff** [comment]
Created entities & bundles as described in https://docs.google.com/document/d/1L3WWBA_Y7HMr4k_GL7PZTrpHj3mWyUO9YsrKBa_whz4/edit#

(this is a new doc, where we're documenting the state of the ESD E3 data management system as it is implemented in code, as opposed to https://docs.google.com/document/d/1_tApwQ24lR0xwd0kbgFjqIKfT9eePxWycD2mE27gOnA/edit# where we have ideas & imaginary things.)

Exported all of @anderoy's field bases from the dev site, so some work is saved in creating all of those fields -- still need to add the field groups & field instances.

Tested to ensure that the user (coach) and related EC can be referenced in views for the contact type -- successful. Everything's ready to be used for site-building!

Big question that isn't quite resolved -- we have a start & end date field in the e3_note entity, but I didn't set up a widget for it yet... what sort of granularity do these input widgets need?

I created a snapshot of the 28-e3entities environment & set up a new testbed for this branch, 29-e3teachers.

Eric, play around at https://29-e3teachers-3mhev6qb5ihtc.us.platform.sh/ -- you can add fields and groups; just be sure to document your additions in https://docs.google.com/document/d/1L3WWBA_Y7HMr4k_GL7PZTrpHj3mWyUO9YsrKBa_whz4/edit# and also leave notes here.

* Add a teacher: https://29-e3teachers-3mhev6qb5ihtc.us.platform.sh/e3-contact/add/teacher
* Add a class assessment: https://29-e3teachers-3mhev6qb5ihtc.us.platform.sh/e3-note/add/class_assessment
* Add an observation log: https://29-e3teachers-3mhev6qb5ihtc.us.platform.sh/e3-note/add/observation_log
* Add a session log: https://29-e3teachers-3mhev6qb5ihtc.us.platform.sh/e3-note/add/session_log
